src/BaselineOfReflectivity/BaselineOfReflectivity.class.st

```diff
@@ -5,6 +5,23 @@ Class {
 	#package : 'BaselineOfReflectivity'
 }
 
+{ #category : 'system annoucements' }
+BaselineOfReflectivity class >> handleMethodModified: aMethodModified [
+
+	aMethodModified oldMethod saveBcToASTCacheWithAST:
+		aMethodModified oldMethod ast
+]
+
+{ #category : 'system annoucements' }
+BaselineOfReflectivity class >> subscribeToMethodModified [
+
+	<script>
+	self codeChangeAnnouncer weak
+		when: MethodModified
+		send: #handleMethodModified:
+		to: self
+]
+
```

Comment:
I am confused as to why it is the baseline class that declares the code change event registration and event handling...

Reply:
I just need to execute at some point when the project is loaded. Do you have any better idea to do that?

Reply:
BaselineOfReflectivity is not a good place for the code, I think (BaseLine is meta data and should not have any code related to execution).

normally we add subscription code like that to a class side #inialize method, they get called on code load.

Reply:
Maybe RFReflectivityASTCacheStrategy could be the place for the code?

Reply:
Yes, I thought that a class-side `initialize` method would be the place, but I didn't know where to put that code as I don't create any class, but only extension methods.

I will try to put that in `RFReflectivityASTCacheStrategy`